faraday/uart/layer_2_protocol.py

```diff
@@ -67,9 +73,14 @@ def run(self):
                     while(not self.serial_tx_queue.empty()):
                         self.ser.write(self.serial_tx_queue.get())
                 #Check for bytes to receive from serial
-                if self.ser.inWaiting() > 0:
-                    rx_buffer_inwaiting = self.ser.inWaiting()
-                    self.serial_rx_queue.put(self.ser.read(rx_buffer_inwaiting))
+                try:
+                    if self.ser.inWaiting() > 0:
+                        rx_buffer_inwaiting = self.ser.inWaiting()
+                        self.serial_rx_queue.put(self.ser.read(rx_buffer_inwaiting))
+                except serial.SerialException as e:
+                    logging.error("USB cable likely disconnected!")
+                    logging.error(e)
+                    sys.exit(1)  # Sys.exit(1) is an error
```

Comment:
@reillyeon is abort() from "os"? Are you suggesting that it would be better because it triggers a core dump? Just trying to educate myself on this subject....